datamap/mv/core/src/main/scala/org/apache/carbondata/mv/rewrite/DefaultMatchMaker.scala

```diff
@@ -448,17 +448,44 @@ object GroupbyGroupbySelectOnlyChildDelta extends DefaultMatchPattern with Predi
       subsumer: ModularPlan,
       compensation: Option[ModularPlan]) = {
     if (subsumee.asInstanceOf[GroupBy].predicateList.contains(exprE)) {
-      if (exprListR.exists(_.semanticEquals(exprE)) || canEvaluate(exprE, exprListR)) true
-      else false
+      if (exprListR.exists(_.semanticEquals(exprE)) || canEvaluate(exprE, exprListR) ||
+          isDerivableForUDF(exprE, exprListR)) {
+        true
+      } else {
+        false
+      }
     } else if (compensation.getOrElse(throw new RuntimeException("compensation cannot be None"))
       .asInstanceOf[Select].predicateList.contains(exprE)) {
-      if (canEvaluate(exprE, exprListR) || exprListR.exists(_.semanticEquals(exprE))) true
-      else false
+      if (canEvaluate(exprE, exprListR) || exprListR.exists(_.semanticEquals(exprE)) ||
+          isDerivableForUDF(exprE, exprListR)) {
+        true
+      } else {
+        false
+      }
     } else {
       false
     }
   }
 
+  /**
+   * org.apache.carbondata.mv.plans.MorePredicateHelper#canEvaluate will be checking the
+   * exprE.references as subset of AttibuteSet(exprListR), which will just take the column name from
+   * UDF, so it will be always false for ScalaUDF.
+   * This method takes care of checking whether the exprE references can be derived form list
+   */
```

Comment:
Please add example explaining about the scenario in comments

Reply:
done